include/cuda/std/detail/libcxx/include/optional

```diff
@@ -999,37 +1058,165 @@ public:
+    {
+      return optional<_Up>(__optional_construct_from_invoke_tag{}, _CUDA_VSTD::forward<_Func>(__f), this->__get());
+    }
+    return optional<_Up>();
+  }
+
+  template<class _Func>
+  _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_AVAILABILITY_THROW_BAD_OPTIONAL_ACCESS
+  constexpr auto transform(_Func&& __f) && {
+    using _Up = remove_cv_t<invoke_result_t<_Func, value_type&&>>;
+    static_assert(!is_array_v<_Up>, "Result of f(std::move(value())) should not be an Array");
+    static_assert(!is_same_v<_Up, in_place_t>,
+                  "Result of f(std::move(value())) should not be std::in_place_t");
+    static_assert(!is_same_v<_Up, nullopt_t>,
+                  "Result of f(std::move(value())) should not be std::nullopt_t");
+    static_assert(is_object_v<_Up>, "Result of f(std::move(value())) should be an object type");
+    if (this->__engaged_)
+    {
+      return optional<_Up>(__optional_construct_from_invoke_tag{}, _CUDA_VSTD::forward<_Func>(__f), _CUDA_VSTD::move(this->__get()));
+    }
+    return optional<_Up>();
+  }
+
+  template<class _Func>
+  _LIBCUDACXX_INLINE_VISIBILITY _LIBCUDACXX_AVAILABILITY_THROW_BAD_OPTIONAL_ACCESS
+  constexpr auto transform(_Func&& __f) const&& {
+    using _Up = remove_cvref_t<invoke_result_t<_Func, const value_type&&>>;
+    static_assert(!is_array_v<_Up>, "Result of f(std::move(value())) should not be an Array");
+    static_assert(!is_same_v<_Up, in_place_t>,
+                  "Result of f(std::move(value())) should not be std::in_place_t");
+    static_assert(!is_same_v<_Up, nullopt_t>,
+                  "Result of f(std::move(value())) should not be std::nullopt_t");
+    static_assert(is_object_v<_Up>, "Result of f(std::move(value())) should be an object type");
+    if (this->__engaged_)
+    {
+      return optional<_Up>(__optional_construct_from_invoke_tag{}, _CUDA_VSTD::forward<_Func>(__f), _CUDA_VSTD::move(this->__get()));
+    }
+    return optional<_Up>();
+  }
+
+  _LIBCUDACXX_TEMPLATE(class _Func)
+    (requires invocable<_Func> _LIBCUDACXX_AND is_copy_constructible_v<value_type>)
+  _LIBCUDACXX_INLINE_VISIBILITY
+  constexpr optional or_else(_Func&& __f) const& {
+    static_assert(is_same_v<remove_cvref_t<invoke_result_t<_Func>>, optional>,
+                  "Result of f() should be the same type as this optional");
+    if (this->__engaged_)
     {
-        return &__x;
+      return *this;
     }
+    return _CUDA_VSTD::forward<_Func>(__f)();
+  }
+
+  _LIBCUDACXX_TEMPLATE(class _Func)
+    (requires invocable<_Func> _LIBCUDACXX_AND is_move_constructible_v<value_type>)
+  _LIBCUDACXX_INLINE_VISIBILITY
+  constexpr optional or_else(_Func&& __f) && {
+    static_assert(is_same_v<remove_cvref_t<invoke_result_t<_Func>>, optional>,
+                  "Result of f() should be the same type as this optional");
+    if (this->__engaged_)
+    {
+      return _CUDA_VSTD::move(*this);
+    }
+    return _CUDA_VSTD::forward<_Func>(__f)();
+  }
+
+    using __base::reset;
```

Comment:
```suggestion
  using __base::reset;
```

Reply:
Actually applied the reverse comment, the block of the monadic functions was indented with 2 spaces, while the rest of this file is 4.